lib/capistrano/recipes/deploy/strategy/jenkins_artifact.rb

```diff
@@ -70,9 +81,8 @@ def deploy!
       }.to_s
     end
 
-    build_num = client.job.get_last_successful_build_number(dir_name)
-    timestamp = client.job.get_build_details(dir_name, build_num)['timestamp']
-    deploy_at = Time.at(timestamp / 1000)
+    last_successful_build = client.job.get_last_successful_build(dir_name)
```

Comment:
Yes, if happened.

Totally we cannot assure any consistency about artifacts without introduction of some sort of giant locking.

But this code may work well in most cases.